data-processing-lib/python/src/data_processing/transform/pipelined_transform.py

```diff
@@ -0,0 +1,202 @@
+    ) -> tuple[list[tuple[bytes, str]], dict[str, Any]]:
+        """
+        Apply the list of transforms to the given datum tuple of filename and byte[]
+        Args:
+            transforms:
+            datum:
+        Returns: same as transform_binary().
+
+        """
+        r_metadata = {}
+        pending_to_process = [(datum[0], datum[1])]
+        for transform in transforms:
+            transform_name = type(transform).__name__
+            to_process = pending_to_process
+            pending_to_process = []
+            for tp in to_process:  # Over all outputs from the last transform  (or the initial input)
+                fname = tp[0]
+                to_transform = tp[1]
+                transformation_tuples, metadata = self._call_transform(transform, to_transform, fname)
+                # Capture the list of outputs from this transform as inputs to the next (or as the return values).
+                for transformation in transformation_tuples:
+                    transformed, extension = transformation
+                    fname = transform_name + "-output" + extension
+                    next = (fname, transformed)
+                    pending_to_process.append(next)
+                # TODO: this is not quite right and might overwrite previous values.
+                # Would be better if we could somehow support lists.
+                r_metadata = r_metadata | metadata
+
+        # Strip the pseudo-base filename from the pending_to_process tuples, leaving only the extension, as required.
+        r_bytes = []
+        for tp in pending_to_process:
+            fname = tp[0]
+            byte_array = tp[1]
+            extension = pathlib.Path(fname).suffix
+            tp = (byte_array, extension)
+            r_bytes.append(tp)
+
+        return r_bytes, r_metadata
+
+    def _call_transform(
+        self,
+        transform: Union[AbstractTableTransform, AbstractBinaryTransform],
+        datum: Union[pa.Table, bytearray],
+        file_name: str,
+    ):
+        is_table_transform = isinstance(datum, pa.Table)
+        if is_table_transform:
+            tables, metadata = transform.transform(datum, file_name)
+            transformation_tuples = []
+            for table in tables:
+                transformation_tuples.append((table, ".parquet"))
+        else:
+            transformation_tuples, metadata = transform.transform_binary(file_name, datum)
+        return transformation_tuples, metadata
+
+    def _call_flush(self, transform: Union[AbstractTableTransform, AbstractBinaryTransform]):
+        is_table_transform = isinstance(transform, AbstractTableTransform)
+        if is_table_transform:
+            tables, metadata = transform.flush()
+            transformation_tuples = []
+            for table in tables:
+                transformation_tuples.append((table, ".parquet"))
+        else:
+            transformation_tuples, metadata = transform.flush_binary()
```

Comment:
This should be the only method called. Why is it more complex that it needs to be

Reply:
Again, trying to avoid byte/Table conversion, but not sure its ready yet.